pypit/par/pypitpar.py

```diff
@@ -0,0 +1,2486 @@
+        descr['overwrite'] = 'Flag to overwrite any existing output files'
+
+        # Instantiate the parameter set
+        super(RunPar, self).__init__(list(pars.keys()),
+                                     values=list(pars.values()),
+                                     defaults=list(defaults.values()),
+                                     options=list(options.values()),
+                                     dtypes=list(dtypes.values()),
+                                     descr=list(descr.values()))
+#                                     , cfg_section='run', cfg_comment='Execution options')
+        self.validate()
+
+    @classmethod
+    def from_dict(cls, cfg):
+        k = cfg.keys()
+        parkeys = [ 'ncpus', 'calcheck', 'calwin', 'setup', 'qa', 'preponly', 'stopcheck',
+                    'useIDname', 'verbosity', 'caldir', 'scidir', 'qadir', 'sortdir', 'overwrite' ]
+        kwargs = {}
+        for pk in parkeys:
+            kwargs[pk] = cfg[pk] if pk in k else None
+        return cls(**kwargs)
+
+    def validate(self):
+        pass
+
+
+class ReducePar(ParSet):
+    """
+    Parameters specific to the reduction procedures used by PypIt.
+    """
+    def __init__(self, spectrograph=None, pipeline=None, detnum=None, masters=None, setup=None,
+                 trim=None, badpix=None, slit_center_frame=None, slit_edge_frame=None,
+                 overscan=None, flatfield=None, flexure=None, wavecalib=None, fluxcalib=None,
+                 skysubtract=None):
+
+        # Grab the parameter names and values from the function
+        # arguments
+        args, _, _, values = inspect.getargvalues(inspect.currentframe())
+        pars = dict([(k,values[k]) for k in args[1:]])      # "1:" to skip 'self'
+
+        # Initialize the other used specifications for this parameter
+        # set
+        defaults = dict.fromkeys(pars.keys())
+        options = dict.fromkeys(pars.keys())
+        dtypes = dict.fromkeys(pars.keys())
+        descr = dict.fromkeys(pars.keys())
+
+        # Fill out parameter specifications.  Only the values that are
+        # *not* None (i.e., the ones that are defined) need to be set
+        defaults['spectrograph'] = 'KECK_LRISb'
+        options['spectrograph'] = ReducePar.valid_spectrographs()
+        dtypes['spectrograph'] = str
+        descr['spectrograph'] = 'Spectrograph that provided the data to be reduced.  ' \
+                                'Options are: {0}'.format(', '.join(options['spectrograph']))
+
+        options['pipeline'] = ReducePar.valid_pipelines()
+        dtypes['pipeline'] = str
+        descr['pipeline'] = 'Pipeline options that pypit can use for reductions.  ' \
+                            'Options are: {0}'.format(', '.join(options['pipeline']))
+
+        dtypes['detnum'] = int
+        descr['detnum'] = 'Restrict reduction to a single detector with this index'
+
+        options['masters'] = ReducePar.allowed_master_options()
```

Comment:
Let's make Masters its own Parset.

See the MasterFrame class for what I had in mind.

And I need to distinguish between Setup A which specifies
the overall setup and setup = A_01_aa which specifies detector
and the specific calibration set.  Maybe

SetupID

and 

setup

Reply:
Let's work this out during the hackday.